json-endpoints/jetpack/class.jetpack-json-api-sync-endpoint.php

```diff
@@ -317,14 +325,29 @@ protected function result() {
 
 		$items = $queue->peek_by_id( $request_body['item_ids'] );
 
-		/** This action is documented in packages/sync/src/modules/Full_Sync.php */
-		$full_sync_module = Modules::get_module( 'full-sync' );
+		// Update Full Sync Status if queue is "full_sync".
+		if ( 'full_sync' === $queue_name ) {
+			$full_sync_module = Modules::get_module( 'full-sync' );
 
-		$full_sync_module->update_sent_progress_action( $items );
+			$full_sync_module->update_sent_progress_action( $items );
+		}
 
 		$buffer = new Queue_Buffer( $request_body['buffer_id'], $request_body['item_ids'] );
 		$response = $queue->close( $buffer, $request_body['item_ids'] );
 
+		// Perform another checkout?
+		if ( $request_body['continue'] ) {
```

Comment:
Trying to trace the code reading through it. Any possibility of 'continue' not being set? Do we need to `isset` or use `empty` instead?

Reply:
:nod: definitely this is worth hardening and will change it to `if ( isset( $request_body['continue'] ) && $request_body['continue']  ) {`

I could see further checks down the road on the explicit value but for now I don't think it makes sense for further validation as the only caller of this is WP.com, any other use will have no impact as it will simply check a buffer out and idle.

Reply:
Cool, let's add that and it gets the green check from me.